bundles/org.eclipse.emfcloud.modelserver.client/src/org/eclipse/emfcloud/modelserver/client/TypedSubscriptionListener.java

```diff
@@ -20,9 +21,19 @@
 public class TypedSubscriptionListener<T> implements NotificationSubscriptionListener<T> {
    private static Logger LOG = LogManager.getLogger(TypedSubscriptionListener.class);
 
-   private final Function<? super String, Optional<T>> updateFunction;
+   private final BiFunction<? super String, ? super String, Optional<? extends T>> updateFunction;
 
-   public TypedSubscriptionListener(final Function<String, Optional<T>> updateFunction) {
+   public TypedSubscriptionListener(final Function<String, Optional<? extends T>> updateFunction) {
+      this(updateFunction, Function.class);
+   }
+
+   public TypedSubscriptionListener(final Function<String, Optional<? extends T>> updateFunction,
+      final Class<?> discriminator) {
```

Comment:
Do we need this `discriminator` here? I can't find a usage yet

Reply:
Hah! Thanks for pointing this out. It Turns Out™ that the `discriminator` is now superfluous.

At one stage, I had two different constructors accepting a `Function` with different type arguments, which erase to the same signature and so needed an additional argument in one of them to distinguish the constructor signatures. But some later refactoring eliminated the second constructor an so now this dummy `discriminator` argument is not needed. Consider it deleted.